### ISSUE TITLE
[QUESTION] Pybind11 module report errors when executing openpose Python API examples09.

### ISSUE THREAD

**Issue by Pinocchioo** [pybind/pybind11#2680]
Pybind11 module report errors when executing openpose Python API examples09：09_keypoints_from_heatmaps.py.The error report is as follows：
```
Starting OpenPose Python Wrapper...
Auto-detecting all available GPUs... Detected 1 GPU(s), using 1 of them starting at GPU 0.
emplaceAndPop(): incompatible function arguments. The following argument types are supported:
    1. (self: openpose.pyopenpose.WrapperPython, arg0: std::vector<std::shared_ptr<op::Datum>, std::allocator<std::shared_ptr<op::Datum> > >) -> bool

Invoked with: <openpose.pyopenpose.WrapperPython object at 0x7fab91920e30>, [<openpose.pyopenpose.Datum object at 0x7fab91920fb8>]

Did you forget to `#include <pybind11/stl.h>`? Or <pybind11/complex.h>,
<pybind11/functional.h>, <pybind11/chrono.h>, etc. Some automatic
conversions are optional and require extra headers to be included
when compiling your pybind11 module.
```
And I don't know why ,I hope someone can answer it. Thank you very much.

**Comment by bstaletic** [comment]
Like the error says:

> Did you forget to `#include <pybind11/stl.h>`?

You did.

**Comment by Pinocchioo** [comment]
> Like the error says:
> 
> > Did you forget to `#include <pybind11/stl.h>`?
> 
> You did.

I'm not running my own program. It's openpose official examples. Do I need to add this header file?

**Comment by Pinocchioo** [comment]
> Like the error says:
> 
> > Did you forget to `#include <pybind11/stl.h>`?
> 
> You did.

I tried to add over file before, but I still reported an error. I just want to ask if anyone has the same mistake as me?

**Comment by YannickJadoul** [comment]
@Pinocchioo, could you give us the example that reproduces this?

**Comment by Pinocchioo** [comment]
> @Pinocchioo, could you give us the example that reproduces this?

After installing the openpose library, I ran the python API example and reported an error. 
cmake-gui: What I modified is：
```
BUILD_PYTHON YES
Add_Entry:
PYTHON_EXECUTABLE = /home/pinocchio/anaconda3/envs/openpose/bin/python3.6m
PYTHON_LIBRARY = 
/home/pinocchio/anaconda3/envs/openpose/lib/libpython3.6m.so
```
Python 3.6.12 installed by CONDA, and the cmake version 3.16. Opencv-Python version 4.4, installed by CONDA.

**Comment by Pinocchioo** [comment]
> @Pinocchioo, could you give us the example that reproduces this?

I didn't make any changes to the code. After the environment was installed, I ran the example.

**Comment by bstaletic** [comment]
https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/7b960082e25872a2cfabc2988c29ea42714c59f4/python/openpose/openpose_python.cpp#L15

So `std::vector<Datum>` can't be implicitly converted from a python list.

https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/7b960082e25872a2cfabc2988c29ea42714c59f4/python/openpose/openpose_python.cpp#L398

You have to populate a `DatumVector`, not a list.

**Comment by YannickJadoul** [comment]
Well, please just link to the code, so we don't have to go chasing it.

Also, did you compile the library yourself? Or installed from a wheel?
At any rate, you'll have to complain to this `openpose` library, if their examples don't work out of the box and if they (possibly) don't use pybind11 correctly?

**Comment by Pinocchioo** [comment]
> Well, please just link to the code, so we don't have to go chasing it.
> 
> Also, did you compile the library yourself? Or installed from a wheel?
> At any rate, you'll have to complain to this `openpose` library, if their examples don't work out of the box and if they (possibly) don't use pybind11 correctly?

I compile the library by cmake —— make -j —— make install the source code that git clone from the openpose git.  And the 09_example :
https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/master/examples/tutorial_api_python/09_keypoints_from_heatmaps.py

**Comment by Pinocchioo** [comment]
> https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/7b960082e25872a2cfabc2988c29ea42714c59f4/python/openpose/openpose_python.cpp#L15
> 
> So `std::vector<Datum>` can't be implicitly converted from a python list.
> 
> https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/7b960082e25872a2cfabc2988c29ea42714c59f4/python/openpose/openpose_python.cpp#L398
> 
> You have to populate a `DatumVector`, not a list.

ok, I will try .

**Comment by Pinocchioo** [comment]
> https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/7b960082e25872a2cfabc2988c29ea42714c59f4/python/openpose/openpose_python.cpp#L15
> 
> So `std::vector<Datum>` can't be implicitly converted from a python list.
> 
> https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/7b960082e25872a2cfabc2988c29ea42714c59f4/python/openpose/openpose_python.cpp#L398
> 
> You have to populate a `DatumVector`, not a list.

I took a look, my source code is the latest code downloaded by git, and the above changes are already included.But it still doesn't work.

**Comment by YannickJadoul** [comment]
@Pinocchioo Again, if the example is broken of a repository that uses pybind11, you'll have to ask them/report to them. We currently have no indication that pyind11 is broken, from your report. Only that this project might not be using it correctly, or needs to make fixes after the 2.6 updates of pybind11?

**Comment by Pinocchioo** [comment]
> @Pinocchioo Again, if the example is broken of a repository that uses pybind11, you'll have to ask them/report to them. We currently have no indication that pyind11 is broken, from your report. Only that this project might not be using it correctly, or needs to make fixes after the 2.6 updates of pybind11?

OK,thank you. This problem in latest openpose V1.7 from git downloading. Now this problem has been solved, after I downloaded the v1.6 relsaseed version of the openpose.
So I think the official examples have some wrong place in latest openpose_python_api example_09.
I have ask the official openpose by https://github.com/CMU-Perceptual-Computing-Lab/openpose/issues/1718.
